### PR TITLE
Refactoring of Earlgrey GPIO driver

### PR DESCRIPTION
**Pull request by mazurek-michal**
### Pull Request Overview

- [x] Add implementation of PadConfig
- [x] Replace `padctl` with `PadConfig` in lowrisc GPIO
- [x] Add trait for supporting board I/O layout configuration
- [x] Pinmux configuration for CW310 board

Depend on changes form: (Merged) https://github.com/tock/tock/pull/3707

### Testing Strategy

Manual test using `CW310` as DUT and `Hyperdebug` with `opentitan` tool
as test bench for generating stimulus. 

### TODO

- [x] rebase on top of master after merging PINMUX
- [ ] Use auto-generated registers for lowrisc GPIO
- [ ] GPIO input filtering support
- [ ] GPIO wakeup/sleep feature

### Documentation Updated

N/A

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

**Comment by bradjc** [comment]
I think a rebase here would help with reviewing now that the register changes are merged.

**Comment by mazurek-michal** [comment]
@cfrantz  @jrvanwhy  @bradjc Cod is rebased and ready for first round of review.

**Review by jrvanwhy — DISMISSED** [review]

File: chips/earlgrey/src/gpio.rs [hunk 1]
```diff
@@ -6,74 +6,141 @@
...
+        }
+    }
 }
 
+/*
+// Configuring first 32 pad as GPIO
 impl<'a> Port<'a> {
     pub const fn new() -> Self {
         Self {
             pins: [
```
Comment (bradjc) [inline chips/earlgrey/src/gpio.rs:95]: Do we need this old code?

File: chips/earlgrey/src/pinmux.rs [hunk 2]
```diff
@@ -205,4 +223,175 @@ impl SelectInput for PinmuxPeripheralIn {
...
+}
+
+impl PadConfig {
+    /// Connect Pad to internal peripheral I/O using pinmux multiplexers
+    pub fn connect(&self) {
+        match *self {
+            PadConfig::Input(pad, peripheral_in) => {
+                peripheral_in.connect_input(PinmuxInsel::from(pad));
+                // default out selector is 0x2 (Hi-Z)
+                // pad.connect_output(peripheral_out);
```
Comment (alistair23) [inline chips/earlgrey/src/pinmux.rs:259]: Should this be removed?

**Comment by mazurek-michal** [comment]
Connected to OpenTitan task: https://github.com/lowRISC/tock/issues/28

File: chips/earlgrey/src/pinmux_config.rs [hunk 3]
```diff
@@ -0,0 +1,40 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2022.
```
Comment (jrvanwhy) [inline chips/earlgrey/src/pinmux_config.rs:3]: Check the year on the copyright -- this should probably be 2023.

**Review by jrvanwhy — APPROVED** [review]

File: chips/earlgrey/src/pinmux_config.rs (hunk 3 above)
Comment (mazurek-michal) [reply chips/earlgrey/src/pinmux_config.rs:3]: Updated.

**Review by jrvanwhy — APPROVED** [review]

**Comment by mazurek-michal** [comment]
@bradjc  Could you check newest version ?

**Review by bradjc — APPROVED** [review]

**Pull request merged**